src/Common/Commands.Common.Authentication.Abstractions/AzureEnvironment.cs

```diff
@@ -229,6 +229,11 @@ public AzureEnvironment(IAzureEnvironment other)
         public string DataLakeEndpointResourceId { get; set; }
 
         /// <summary>
+        /// The token audience required for communicating with the Batch service in this enviornment
+        /// </summary>
+        public string BatchEndpointResourceId { get; set; }
```

Comment:
This is a breaking change to an interface that must never break.  We are also not adding any more new values to the environment - there needs to be a different mechanism for discovering this value, and we will need to store it in an extended property.

Reply:
How is adding a property a breaking change? Because it's an interface?

> there needs to be a different mechanism for discovering this value, and we will need to store it in an extended property

Can you expand a bit more on this? We followed what looked to be "the pattern" here -- how should this work?

Reply:
Pinging on this - @markcowl can you tell me what I should be doing here...?